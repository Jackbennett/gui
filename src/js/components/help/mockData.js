export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
