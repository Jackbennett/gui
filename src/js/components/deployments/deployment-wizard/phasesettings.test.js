import React from 'react';
import renderer from 'react-test-renderer';
import PhaseSettings from './phasesettings';
import { undefineds } from '../../../../../tests/mockData';

describe('PhaseSettings Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PhaseSettings phases={[{ batch_size: 0 }]} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
