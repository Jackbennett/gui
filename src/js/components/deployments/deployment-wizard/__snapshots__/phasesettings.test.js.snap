// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseSettings Component renders correctly 1`] = `
<div>
  <div>
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          />
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          >
            Batch size
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          >
            Phase begins
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          >
            Delay before next phase
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <div
              class="MuiChip-root MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                Phase 1
              </span>
            </div>
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            100
            <span
              class="info"
              style="margin-left: 5px;"
            >
              (1 device)
            </span>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <time
              datetime="2019-01-01T13:00:00+00:00"
            >
              2019-01-01 13:00
            </time>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            -
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          />
        </tr>
      </tbody>
    </table>
    <div
      class="MuiButtonBase-root MuiChip-root margin-top-small MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-icon MuiChip-iconColorPrimary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiChip-label"
      >
        Add a phase
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;
